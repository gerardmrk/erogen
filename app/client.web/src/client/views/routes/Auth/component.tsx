import * as React from "react";
import styles from "./component.styles.scss";
import HeadTags from "@client/views/components/HeadTags";
import Container from "@client/views/components/ui-elements/Container";
import { LocalProps } from ".";

export type Props = LocalProps;

export type State = {};

export class Auth extends React.PureComponent<Props, State> {
  public static i18nNamespace = "route_Auth";

  public render() {
    return (
      <React.Fragment>
        <HeadTags
          path={this.props.path}
          title={this.props.t("title")}
          description={this.props.t("description")}
          metaType={this.props.metaType}
          metaImgPath={this.props.metaImgPath}
          metaImgAlt={this.props.metaImgAlt}
          metaTwitterCardType={this.props.metaTwitterCardType}
        />

        <div className={styles.main}>
          <Container>
            <h1>{"Auth"}</h1>
          </Container>
        </div>
      </React.Fragment>
    );
  }
}
