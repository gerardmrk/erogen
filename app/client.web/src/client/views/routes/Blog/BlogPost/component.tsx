import * as React from "react";
import styles from "./component.styles.scss";

type Props = {};

type State = {};

export class BlogPost extends React.PureComponent<Props, State> {
  public static readonly chunkName = "Blog-BlogPost";
  public static readonly i18nNamespace = "";

  public render() {
    return (
      <div className={styles.main}>
        <h1>{"BlogPost"}</h1>
      </div>
    );
  }
}

export default BlogPost;
