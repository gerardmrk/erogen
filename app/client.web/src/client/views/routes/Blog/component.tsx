import * as React from "react";
import styles from "./component.styles.scss";
import { Switch } from "react-router-dom";
import { LocalProps } from ".";
import EnhancedRoute from "@client/views/components/EnhancedRoute";
import Container from "@client/views/components/ui.elements/Container";
import HeadTags from "@client/views/components/HeadTags";

export type Props = LocalProps;

export type State = {};

export class Blog extends React.PureComponent<Props, State> {
  public static i18nNamespace = "route_Blog";

  public render() {
    return (
      <React.Fragment>
        <HeadTags
          path={this.props.path}
          title={this.props.t("title")}
          description={this.props.t("description")}
          metaType={this.props.metaType}
          metaImgPath={this.props.metaImgPath}
          metaImgAlt={this.props.metaImgAlt}
          metaTwitterCardType={this.props.metaTwitterCardType}
        />

        <div className={styles.main}>
          <Container>
            <h1>{"Blog"}</h1>

            <Switch>
              {this.props.routes.map((r, i) => (
                <EnhancedRoute key={i} {...r} />
              ))}
            </Switch>
          </Container>
        </div>
      </React.Fragment>
    );
  }
}
