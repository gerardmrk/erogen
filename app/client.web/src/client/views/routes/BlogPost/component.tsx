import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps } from ".";
import HeadTags from "@client/views/components/HeadTags";

type Props = LocalProps;

type State = {};

export class BlogPost extends React.PureComponent<Props, State> {
  public static i18nNamespace = "route_BlogPost";

  public render() {
    return (
      <React.Fragment>
        <HeadTags
          path={this.props.path}
          title={this.props.t("title")}
          description={this.props.t("description")}
          metaType={this.props.metaType}
          metaImgPath={this.props.metaImgPath}
          metaImgAlt={this.props.metaImgAlt}
          metaTwitterCardType={this.props.metaTwitterCardType}
        />

        <div className={styles.main}>
          <h1>{"BlogPost"}</h1>
        </div>
      </React.Fragment>
    );
  }
}
