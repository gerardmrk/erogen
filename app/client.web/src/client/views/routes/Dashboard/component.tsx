import * as React from "react";
import styles from "./component.styles.scss";

type Props = {};

type State = {};

export class Dashboard extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <h1>{"Dashboard"}</h1>
      </div>
    );
  }
}
