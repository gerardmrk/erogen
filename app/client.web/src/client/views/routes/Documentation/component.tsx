import * as React from "react";
import { Heading } from "evergreen-ui/esm/typography";
import styles from "./styles.scss";
import Container from "@client/views/components/Container";

type Props = {};

type State = {};

export class Documentation extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <Container>
          <Heading size={700}>{"Documentation"}</Heading>
        </Container>
      </div>
    );
  }
}

export default Documentation;
