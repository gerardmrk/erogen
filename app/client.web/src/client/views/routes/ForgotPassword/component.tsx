import * as React from "react";
import { Heading } from "evergreen-ui";
import styles from "./styles.scss";

type Props = {};

type State = {};

export class ForgotPassword extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <Heading size={700}>{"ForgotPassword"}</Heading>
      </div>
    );
  }
}

export default ForgotPassword;
