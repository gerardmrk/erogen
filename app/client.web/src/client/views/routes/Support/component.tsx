import * as React from "react";
import { Heading } from "evergreen-ui";
import styles from "./styles.scss";

type Props = {};

type State = {};

export class Support extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <Heading size={700}>{"Support"}</Heading>
      </div>
    );
  }
}

export default Support;
