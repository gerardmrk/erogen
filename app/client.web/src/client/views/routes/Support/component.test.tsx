import * as React from "react";
import { shallow } from "enzyme";
import { Support } from "./component";

describe("<Support/>", () => {
  it("renders ok", () => {
    const wrapper = shallow(<Support />);
    expect(wrapper).toExist();
    expect(wrapper).toHaveClassName("main");
  });

  it("has static properties required for a route", () => {
    expect(Support).toHaveProperty("chunkName");
    expect(Support).toHaveProperty("i18nNamespace");
  });
});
