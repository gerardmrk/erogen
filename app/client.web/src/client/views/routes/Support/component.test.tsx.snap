// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Support/> snapshot 1`] = `
<div
  className="main"
>
  <div
    className="ui container"
  >
    <h1>
      Support
    </h1>
  </div>
</div>
`;
